src/main/scala/eventstore/PersistentSubscriptionActor.scala

```diff
@@ -45,6 +25,20 @@ object PersistentSubscriptionActor {
       autoAck
     ))
   }
+
+  sealed trait State
+
+  private case object Unsubscribed extends State
+
+  private case object LiveProcessing extends State
+
+  private case object CatchingUp extends State
+
+  sealed trait Data
+  private final case class ConnectionDetails()
```

Comment:
Whoops, my bad. I was refactoring before I'd had my coffee. Correcting that now.